ports-of-call/portable_arrays.hpp

```diff
@@ -26,73 +26,145 @@
 
 #include "portability.hpp"
 #include <algorithm>
+#include <array>
 #include <assert.h>
 #include <cstddef> // size_t
 #include <cstring> // memset()
+#include <functional>
+#include <numeric>
+#include <type_traits>
 #include <utility> // swap()
 
+// maximum number of dimensions
+constexpr std::size_t MAXDIM = 6;
+// array type of dimensions/strides
+using narr = std::array<std::size_t, MAXDIM>;
```

Comment:
don't love that this is lower-case. I'd find this type easier to interpret if it were, e.g.,
```suggestion
using Narr_t = std::array<std::size_t, MAXDIM>;
```